orion-server/src/main/java/com/pinterest/orion/core/kafka/KafkaCluster.java

```diff
@@ -171,27 +174,35 @@ public void addNodeWithoutAgent(NodeInfo info) {
 
   @JsonIgnore
   public Map<String, KafkaTopicDescription> getTopicDescriptionFromKafka() throws InterruptedException,
-                                                                           ExecutionException {
+                                                                                  ExecutionException, TimeoutException {
+    return getTopicDescriptionFromKafka(DEFAULT_METADATA_TIMEOUT_MS);
```

Comment:
Can this also use user's input for the timeout? By default 30 sec is okay, but I think ideally this should use the custom value use provides.

Reply:
@vahidhashemian this is an overloaded method that would use the default timeout. there is a method right beneath this one where it can take custom values.